src/Compilers/CSharp/Test/Semantic/Diagnostics/GetDiagnosticsTests.cs

```diff
@@ -694,5 +695,43 @@ public async Task TestGetAnalyzerSyntaxDiagnosticsWithCancellation(bool concurre
             var diagnostic = Assert.Single(diagnostics);
             Assert.Equal(RegisterSyntaxTreeCancellationAnalyzer.DiagnosticId, diagnostic.Id);
         }
+
+        [Fact]
+        public async Task TestEventQueuePartialCompletionForSpanBasedQuery()
```

Comment:
💡 Your explanation in the pull request is particularly clear. Consider adding it as a documentation comment above this method.